### PR TITLE
WGWService Recordings phase 2 

### PR DESCRIPTION
**Pull request by eran-braun**
* Changed file names to be saved with timestamp
* added saving a json file describing the recording in a diffrent  folder RecJSON

### CODE REVIEW

File: plugins/janus_videoroom.c [hunk 1]
```diff
@@ -7388,6 +7390,15 @@ void janus_videoroom_slow_link(janus_plugin_session *handle, int uplink, int vid
 	janus_refcount_decrease(&session->ref);
 }
 
+char *get_time_stamp()
+{
+	struct timeval current_time;
```
Comment (shmuel7) [inline plugins/janus_videoroom.c]: initialization

File: plugins/janus_videoroom.c [hunk 2]
```diff
@@ -7388,6 +7390,15 @@ void janus_videoroom_slow_link(janus_plugin_session *handle, int uplink, int vid
...
+	gettimeofday(&current_time, NULL);
```
Comment (shmuel7) [inline plugins/janus_videoroom.c:7396]: validate current_time return value

File: plugins/janus_videoroom.c [hunk 3]
```diff
@@ -7388,6 +7390,15 @@ void janus_videoroom_slow_link(janus_plugin_session *handle, int uplink, int vid
...
+	char *timestamp;
+	asprintf(&timestamp, "%ld_%ld", current_time.tv_sec, current_time.tv_usec);
+	return timestamp;
```
Comment (shmuel7) [inline plugins/janus_videoroom.c:7399]: validate return value timestamp

File: plugins/janus_videoroom.c [hunk 4]
```diff
@@ -7398,8 +7409,10 @@ static void janus_videoroom_recorder_create(janus_videoroom_publisher *participa
 		memset(filename, 0, 255);
 		if (participant->recording_base)
 		{
+			char *timestamp = get_time_stamp();
+
 			/* Use the filename and path we have been provided */
-			g_snprintf(filename, 255, "%s-audio", participant->recording_base);
+			g_snprintf(filename, 255, "%s-audio-%s", participant->recording_base,timestamp);
```
Comment (shmuel7) [inline plugins/janus_videoroom.c]: check timestamp (null)

File: plugins/janus_videoroom.c [hunk 5]
```diff
@@ -7433,8 +7446,9 @@ static void janus_videoroom_recorder_create(janus_videoroom_publisher *participa
 		memset(filename, 0, 255);
 		if (participant->recording_base)
 		{
+			char *timestamp = get_time_stamp();
 			/* Use the filename and path we have been provided */
-			g_snprintf(filename, 255, "%s-video", participant->recording_base);
+			g_snprintf(filename, 255, "%s-video-%s", participant->recording_base, timestamp);
```
Comment (shmuel7) [inline plugins/janus_videoroom.c]: check timestamp (null)

File: plugins/janus_videoroom.c [hunk 6]
```diff
@@ -7489,14 +7503,41 @@ static void janus_videoroom_recorder_create(janus_videoroom_publisher *participa
 		participant->drc = rc;
 	}
 }
+// CARBYNE-S3-UPLOAD
+
+int create_recording_json(char *filename, char *codec)
+{
+	char *jsonObj;
+	asprintf(&jsonObj, "{\n \"recordingName\" : \"%s\" ,\n \"directory\" : \"/home/ubuntu/recordings/\" ,\n\"codec\" : \"%s\"\n}", filename, codec);
```
Comment (shmuel7) [inline plugins/janus_videoroom.c]: Use defines for /home/ubuntu/recordings/ and for the whole json structure

File: plugins/janus_videoroom.c [hunk 7]
```diff
@@ -7489,14 +7503,41 @@ static void janus_videoroom_recorder_create(janus_videoroom_publisher *participa
...
+	char *fullFileName;
+	// removeSubstr(filename,".mjr");
+	asprintf(&fullFileName, "/home/ubuntu/RecJSON/%s.json", filename);
+	printf("[file debug]------%s\n", fullFileName);
```
Comment (shmuel7) [inline plugins/janus_videoroom.c]: Check fullFileName not Null before using it

File: plugins/janus_videoroom.c [hunk 8]
```diff
@@ -7489,14 +7503,41 @@ static void janus_videoroom_recorder_create(janus_videoroom_publisher *participa
...
+	FILE *fp;
+
+	fp = fopen(fullFileName, "w+");
 
+	fputs(jsonObj, fp);
```
Comment (shmuel7) [inline plugins/janus_videoroom.c]: validate fp before using it

File: plugins/janus_videoroom.c (hunk 7 above)
Comment (shmuel7) [inline plugins/janus_videoroom.c]: Does printf go to wgw logs? cloud watch?

File: plugins/janus_videoroom.c [hunk 9]
```diff
@@ -7489,14 +7503,41 @@ static void janus_videoroom_recorder_create(janus_videoroom_publisher *participa
...
 static void janus_videoroom_recorder_close(janus_videoroom_publisher *participant)
 {
 	if (participant->arc)
 	{
 		janus_recorder *rc = participant->arc;
 		participant->arc = NULL;
 		janus_recorder_close(rc);
+		// upload file
+		// upload_recording(rc->filename,rc->codec);
+		printf("[file debug]------calling create_recording_json\n");
```
Comment (shmuel7) [inline plugins/janus_videoroom.c:7555]: Does printf go to wgw logs? cloud watch?

Comment (shmuel7) [inline plugins/janus_videoroom.c:7555]: Does printf go to wgw logs? cloud watch?

File: plugins/janus_videoroom.c [hunk 10]
```diff
@@ -7505,6 +7546,14 @@ static void janus_videoroom_recorder_close(janus_videoroom_publisher *participan
 		janus_recorder *rc = participant->vrc;
 		participant->vrc = NULL;
 		janus_recorder_close(rc);
+		// upload file
+		// upload_recording(rc->filename,rc->codec);
+		// try watch
+		printf("[file debug]------calling create_recording_json\n");
```
Comment (shmuel7) [inline plugins/janus_videoroom.c:7568]: Does printf go to wgw logs? cloud watch?

File: plugins/janus_videoroom.c [hunk 11]
```diff
@@ -7505,6 +7546,14 @@ static void janus_videoroom_recorder_close(janus_videoroom_publisher *participan
 		janus_recorder *rc = participant->vrc;
 		participant->vrc = NULL;
 		janus_recorder_close(rc);
+		// upload file
+		// upload_recording(rc->filename,rc->codec);
```
Comment (shmuel7) [inline plugins/janus_videoroom.c]: remove unnecessary comments

File: plugins/janus_videoroom.c [hunk 12]
```diff
@@ -7505,6 +7546,14 @@ static void janus_videoroom_recorder_close(janus_videoroom_publisher *participan
...
+		// try watch
+		printf("[file debug]------calling create_recording_json\n");
+
+		int res = create_recording_json(rc->filename, rc->codec);
+		printf("[file debug]------res = %d\n", res);
```
Comment (shmuel7) [inline plugins/janus_videoroom.c:7571]: Does printf go to wgw logs? cloud watch?

**Review by shmuel7 — CHANGES_REQUESTED** [review]

File: plugins/janus_videoroom.c (hunk 12 above)
Comment (eran-braun) [reply plugins/janus_videoroom.c:7571]: yes

File: plugins/janus_videoroom.c (hunk 10 above)
Comment (eran-braun) [reply plugins/janus_videoroom.c:7568]: yes

File: plugins/janus_videoroom.c (hunk 9 above)
Comment (eran-braun) [reply plugins/janus_videoroom.c:7555]: yes

Comment (eran-braun) [reply plugins/janus_videoroom.c:7555]: yes

File: plugins/janus_videoroom.c (hunk 7 above)
Comment (eran-braun) [reply plugins/janus_videoroom.c]: yes

File: plugins/janus_videoroom.c (hunk 2 above)
Comment (eran-braun) [reply plugins/janus_videoroom.c:7396]: that is the init process for current_time,

File: plugins/janus_videoroom.c (hunk 1 above)
Comment (eran-braun) [reply plugins/janus_videoroom.c]: init on line 7396

File: plugins/janus_videoroom.c [hunk 13]
```diff
@@ -7388,6 +7390,17 @@ void janus_videoroom_slow_link(janus_plugin_session *handle, int uplink, int vid
 	janus_refcount_decrease(&session->ref);
 }
 
+char *get_time_stamp()
+{
+	struct timeval current_time;
+	gettimeofday(&current_time, NULL);
+	char *timestamp;
```
Comment (shmuel7) [inline plugins/janus_videoroom.c]: Initialize to null

**Review by shmuel7 — APPROVED** [review]

**Pull request merged**